week06/my-app/src/app/list/ListItem.tsx

```diff
@@ -0,0 +1,42 @@
+'use client';
+
+import type Post from '@/models/post';
+import type { WithId } from 'mongodb';
+import Link from 'next/link';
+
+export default function ListItem({ result }: { result: WithId<Post>[] }) {
+  return (
+    <div className='list-bg'>
+      {result.map((_, i) => (
+        <div className='list-item' key={i}>
+          <Link href={'/detail/' + result[i]._id}>
+            <h4>{result[i].title}</h4>
+          </Link>
+          <Link href={'/edit/' + result[i]._id}>✏️</Link>
+          <span className="delete-button"
+            onClick={(e) => {
+              fetch('/api/post/delete', {
+                method: 'DELETE',
+                body: result[i]._id.toString(),
+              })
```

Comment:
- **문제 원인**
여기서 body를 보면, `body: result[i]._id.toString()` 으로, body에는 문자열이 담깁니다. 그래서 이후 api/post/delete 코드에서 req.body로 받아온 값 또한 문자열이 됩니다.

Reply:
- **수정방법 1: delete 코드만 수정)** 이 코드에선 수정 필요 없음, delete 수정 부분만 확인

Reply:
- **수정방법 2: ListItem, delete 코드 둘 다 수정)**
 클라이언트 측에서 객체 형태로 반환하도록 수정할 수 있습니다. 
```suggestion
            onClick={(e) => {
              fetch('/api/post/delete', {
                method: 'DELETE',
                headers: { 
                  'Content-Type': 'application/json', 
                },
                body: JSON.stringify({
                  _id: result[i]._id.toString(),
                  // author: result[i].author  // 필요 시 추가 가능. 
                  // delete에서 req.body를 받는것에 맞춰서 작성!
                }),
              })
```